driver-core/src/test/functional/com/mongodb/client/model/expressions/ArithmeticExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,233 @@
+        assertThrows(IllegalArgumentException.class, () -> of((Decimal128) null));
+
+        // expression equality differs from bson equality
+        assertExpression(true, of(1L).eq(of(1.0)));
+        assertExpression(true, of(1L).eq(of(1)));
+
+        // bson equality; underlying type is preserved
+        // this behaviour is not defined by the API, but tested for clarity
+        assertEquals(toBsonValue(1), evaluate(of(1)));
+        assertEquals(toBsonValue(1L), evaluate(of(1L)));
+        assertEquals(toBsonValue(1.0), evaluate(of(1.0)));
+        assertNotEquals(toBsonValue(1), evaluate(of(1L)));
+        assertNotEquals(toBsonValue(1.0), evaluate(of(1L)));
+    }
+
+    @Test
+    public void multiplyTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/multiply/
+        assertExpression(
+                2.0 * 2,
+                of(2.0).multiply(of(2)),
+                "{'$multiply': [2.0, 2]}");
+
+        // mixing integers and numbers
+        IntegerExpression oneInt = of(1);
+        NumberExpression oneNum = of(1.0);
+        IntegerExpression resultInt = oneInt.multiply(oneInt);
+        NumberExpression resultNum = oneNum.multiply(oneNum);
+        // compile time error if these were IntegerExpressions:
+        NumberExpression r2 = oneNum.multiply(oneInt);
+        NumberExpression r3 = oneInt.multiply(oneNum);
+        assertExpression(1, resultInt);
+        // 1 is also a valid expected value in our API
+        assertExpression(1.0, resultNum);
+        assertExpression(1.0, r2);
+        assertExpression(1.0, r3);
+
+        // convenience
+        assertExpression(2.0, of(1.0).multiply(2.0));
+        assertExpression(2, of(1).multiply(2));
+    }
+
+    @Test
+    public void divideTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/divide/
+        assertExpression(
+                1.0 / 2.0,
+                of(1.0).divide(of(2.0)),
+                "{'$divide': [1.0, 2.0]}");
+        // unlike Java's 1/2==0, dividing any type of numbers always yields an
+        // equal result, in this case represented using a double.
+        assertExpression(
+                0.5,
+                of(1).divide(of(2)),
+                "{'$divide': [1, 2]}");
+
+        // however, there are differences in evaluation between numbers
+        // represented using Decimal128 and double:
+        assertExpression(
+                2.5242187499999997,
+                of(3.231).divide(of(1.28)));
+        assertExpression(
+                Decimal128.parse("2.52421875"),
+                of(Decimal128.parse("3.231")).divide(of(Decimal128.parse("1.28"))));
+        assertExpression(
+                Decimal128.parse("2.52421875"),
+                of(Decimal128.parse("3.231")).divide(of(1.28)));
+        assertExpression(
+                Decimal128.parse("2.52421875"),
+                of(3.231).divide(of(Decimal128.parse("1.28"))));
```

Comment:
This fails for me (MongoDB 6.0.2, Liberica JDK 17.0.3):
```
org.opentest4j.AssertionFailedError: 
Expected :BsonDecimal128{value=2.52421875}
Actual   :BsonDecimal128{value=2.524218750000}
```

Reply:
Failed for me as well. Updated.